arch/lkl/kernel/setup.c

```diff
@@ -43,6 +43,9 @@ int run_init_process(const char *init_filename)
 
 static void __init lkl_run_kernel(void *arg)
 {
+	/* Nobody will ever join us */
+	lkl_ops->thread_detach();
```

Comment:
I think it matches the semantics --- on POSIX you do it to save memory, on Windows the memory is saved for you already. Like a memory barrier that becomes a no-op on an architecture that guarantees reads/writes are not reordered. The memory barrier is still there, it's just that all operations have a memory barrier between them. I'd say a NULL `thread_detach` should be reserved for a platform where `thread_create` does acquire memory, but it's not possible to release it.

What do you think? If you disagree, I'll switch it over.